### PR TITLE
Add persistent authentication

### PR DESCRIPTION
**Pull request by krconv**
The application now stores a refresh token after the user logs in, so the access can be regranted without requiring the user to login again. Now the applications starts at the login activity and ensures that the session is authenticated with a request to the REST API. Also fixes some bugs in the Login activity.

### CODE REVIEW

File: app/src/main/java/org/fundacionparaguaya/advisorapp/data/remote/UserLoginTask.java [hunk 1]
```diff
@@ -12,37 +12,50 @@
...
     protected Boolean doInBackground(Void... voids) {
         try {
-            Response<LoginIr> response = familyService
-                    .login("Basic YmFyQ2xpZW50SWRQYXNzd29yZDpzZWNyZXQ=", user.getUsername(), user.getPassword()).execute();
+            Response<LoginIr> response = null;
+            User user = manager.getUser();
+            if (user.getLogin().getRefreshToken() != null) {
+                 response = manager.getAuthService()
+                        .loginWithRefreshToken(
+                                "Basic YmFyQ2xpZW50SWRQYXNzd29yZDpzZWNyZXQ=",
```
Comment (bhylak) [inline app/src/main/java/org/fundacionparaguaya/advisorapp/data/remote/UserLoginTask.java]: put in strings.xml file (set untranslatable to true)

File: app/src/main/java/org/fundacionparaguaya/advisorapp/dependencyinjection/DatabaseModule.java [hunk 2]
```diff
@@ -36,12 +38,12 @@ public DatabaseModule(Application application) {
                 "Advisor.db"
         ).build();
 
-
         Retrofit retrofit = new Retrofit.Builder()
                 .baseUrl("http://povertystoplightiqp.org:8080/")
```
Comment (bhylak) [inline app/src/main/java/org/fundacionparaguaya/advisorapp/dependencyinjection/DatabaseModule.java]: change http://povertystoplightiqp.org:8080/ to strings.xml

File: app/src/main/java/org/fundacionparaguaya/advisorapp/fragments/LoginFragment.java [hunk 3]
```diff
@@ -147,21 +166,118 @@ private void attemptLogin() {
...
+
+    PasswordLoginTask(LoginFragment loginFragment) {
+        super(loginFragment);
+    }
+
+    @Override
+    protected Response<LoginIr> login(User user) throws IOException {
+        return mAuthManager.getAuthService()
+                .loginWithPassword(
+                        "Basic YmFyQ2xpZW50SWRQYXNzd29yZDpzZWNyZXQ=",
```
Comment (bhylak) [inline app/src/main/java/org/fundacionparaguaya/advisorapp/fragments/LoginFragment.java]: also here

File: app/src/main/java/org/fundacionparaguaya/advisorapp/fragments/LoginFragment.java [hunk 4]
```diff
@@ -147,21 +166,118 @@ private void attemptLogin() {
...
+class RefreshTokenLoginTask extends AbstractLoginTask {
+
+    RefreshTokenLoginTask(LoginFragment loginFragment) {
+        super(loginFragment);
+    }
+    @Override
+    protected Response<LoginIr> login(User user) throws IOException {
+        return mAuthManager.getAuthService()
+                .loginWithRefreshToken(
+                        "Basic YmFyQ2xpZW50SWRQYXNzd29yZDpzZWNyZXQ=",
```
Comment (bhylak) [inline app/src/main/java/org/fundacionparaguaya/advisorapp/fragments/LoginFragment.java]: and here

File: app/src/main/java/org/fundacionparaguaya/advisorapp/data/remote/UserLoginTask.java (hunk 1 above)
Comment (krconv) [reply app/src/main/java/org/fundacionparaguaya/advisorapp/data/remote/UserLoginTask.java]: Stored it as a static final string

File: app/src/main/java/org/fundacionparaguaya/advisorapp/dependencyinjection/DatabaseModule.java (hunk 2 above)
Comment (krconv) [reply app/src/main/java/org/fundacionparaguaya/advisorapp/dependencyinjection/DatabaseModule.java]: Stored it as a static final string

File: app/src/main/java/org/fundacionparaguaya/advisorapp/fragments/LoginFragment.java (hunk 3 above)
Comment (krconv) [reply app/src/main/java/org/fundacionparaguaya/advisorapp/fragments/LoginFragment.java]: Stored it as a static final string

File: app/src/main/java/org/fundacionparaguaya/advisorapp/fragments/LoginFragment.java (hunk 4 above)
Comment (krconv) [reply app/src/main/java/org/fundacionparaguaya/advisorapp/fragments/LoginFragment.java]: Stored it as a static final string

**Pull request merged**